codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/waiters/RustJmespathShapeTraversalGenerator.kt

```diff
@@ -429,6 +431,40 @@ class RustJmespathShapeTraversalGenerator(
                 }
             }
 
+            "keys" -> {
+                if (expr.arguments.size != 1) {
+                    throw InvalidJmesPathTraversalException("Keys function takes exactly one argument")
+                }
+                val arg = generate(expr.arguments[0], bindings)
+                if (!arg.isObject()) {
+                    throw InvalidJmesPathTraversalException("Argument to `keys` function must be an object type")
+                }
+                GeneratedExpression(
+                    identifier = ident,
+                    outputType = RustType.Vec(RustType.String),
+                    outputShape = TraversedShape.Array(null, TraversedShape.String(null)),
+                    output =
+                        writable {
+                            arg.output(this)
+                            val out = arg.outputShape.shape
+                            when (out) {
```

Comment:
`out` is a special word in Kotlin per https://kotlinlang.org/docs/keyword-reference.html#modifier-keywords
We should change it to `output` or similar.

Reply:
Fixed in https://github.com/smithy-lang/smithy-rs/pull/3755/commits/f964801602a10eb7de34244b28544b32fc35ffaf